lib/visenze_tracker.dart

```diff
@@ -40,33 +39,36 @@ class VisenzeTracker {
     _sessionManager.userId = uid;
   }
 
+  /// Return remaining time for session
+  int get sessionTimeRemaining {
+    return _sessionManager.sessionTimeRemaining;
+  }
+
+  /// Reset the current session and return the new sessionId
+  String resetSession() {
+    return _sessionManager.resetSession();
+  }
+
   /// Send a request to ViSenze analytics server with event name [action] and provided [queryParams]
-  ///
-  /// Execute [onSuccess] on request success and [onError] on request error
-  Future<void> sendEvent(String action, Map<String, dynamic> queryParams,
-      {void Function()? onSuccess, void Function(String err)? onError}) async {
+  Future<void> sendEvent(
+      String action, Map<String, dynamic> queryParams) async {
     var trackingData = await _getTrackerParams(action, queryParams);
     Uri url = Uri.https(
         _useStaging ? _stagingEndpoint : _endpoint, _path, trackingData);
-
-    var response = await http.get(url);
-    if (response.statusCode == 200 && onSuccess != null) {
-      onSuccess();
-    } else if (response.statusCode != 200 && onError != null) {
-      Map<String, dynamic> body = jsonDecode(response.body);
-      onError(body['error']['message']);
+    final resp = await http.get(url);
+    if (resp.statusCode != 200) {
+      return Future.error(resp.body);
     }
   }
 
   /// Send batch request to ViSenze analytics server with event name [action] and params list [queryParamsList]
-  ///
-  /// Execute [onSuccess] on each request success and [onError] on each request error
   Future<void> sendEvents(
-      String action, List<Map<String, dynamic>> queryParamsList,
-      {void Function()? onSuccess, void Function(String err)? onError}) async {
+      String action, List<Map<String, dynamic>> queryParamsList) async {
+    final List<Future> futures = [];
     for (final params in queryParamsList) {
-      sendEvent(action, params, onSuccess: onSuccess, onError: onError);
+      futures.add(sendEvent(action, params));
```

Comment:
need to check if event is `transaction`.

if it is `transaction`, need to check if the `transId` is provided.

if `transId` is not provided, need to auto generate it and append as part of the param.

you can refer to this as an example:
https://github.com/visenze/visenze-tracking-javascript/blob/master/tracking/src/Tracker.js#L140